tests/test_entities_generic_handler.py

```diff
@@ -0,0 +1,2945 @@
+            }
+        )
+
+    def _mock_jira_event(self, jira_issue, jira_event, jira_worklog=None, jira_comment=None):
+        """Helper method to mock Jira issue event."""
+
+        mocked_jira_event = copy.deepcopy(jira_event)
+
+        if jira_worklog:
+            mocked_jira_event["worklog"]["issueId"] = jira_issue.key
+            mocked_jira_event["worklog"]["id"] = jira_worklog.id
+        elif jira_comment:
+            mocked_jira_event["comment"]["id"] = jira_comment.id
+            mocked_jira_event["issue"]["id"] = jira_issue.key
+            mocked_jira_event["issue"]["key"] = jira_issue.key
+        else:
+            mocked_jira_event["issue"] = {
+                "id": jira_issue.key,
+                "key": jira_issue.key
+            }
+        return mocked_jira_event
+
+    def _check_jira_issue(self, bridge, sg_entity, sync_in_fptr=None):
+        """Helper method to check that a Jira issue is correctly created."""
+
+        # Jira Issue should be created
+        jira_issue = bridge.jira.issue(sg_entity[SHOTGUN_JIRA_ID_FIELD])
+        self.assertIsNotNone(jira_issue)
+
+        # its "Sync in FPTR" field should be set to "True"
+        sync_in_fptr_jira_field_id = bridge.jira.get_jira_issue_field_id(JIRA_SYNC_IN_FPTR_FIELD.lower())
+        self.assertEqual(jira_issue.get_field(sync_in_fptr_jira_field_id).value, sync_in_fptr)
+
+        # all its FPTR fields should be filled with the Task data
+        sg_type_jira_field_id = bridge.jira.get_jira_issue_field_id(JIRA_SHOTGUN_TYPE_FIELD.lower())
+        self.assertEqual(jira_issue.get_field(sg_type_jira_field_id), sg_entity["type"])
+
+        sg_id_jira_field_id = bridge.jira.get_jira_issue_field_id(JIRA_SHOTGUN_ID_FIELD.lower())
+        self.assertEqual(jira_issue.get_field(sg_id_jira_field_id), str(sg_entity["id"]))
+
+        sg_url_jira_field_id = bridge.jira.get_jira_issue_field_id(JIRA_SHOTGUN_URL_FIELD.lower())
+        sg_entity_url = bridge.shotgun.get_entity_page_url(sg_entity)
+        self.assertEqual(jira_issue.get_field(sg_url_jira_field_id), sg_entity_url)
+
+        return jira_issue
+
+# Mock Flow Production Tracking with mockgun, this works only if the code uses shotgun_api3.Shotgun
+# and does not `from shotgun_api3 import Shotgun` and then `sg = Shotgun(...)`
+@mock.patch("shotgun_api3.Shotgun")
+class TestEntitiesGenericHandlerSettings(TestEntitiesGenericHandler):
+    """Test the configuration settings for the Entities Generic Handler."""
+
+    def test_bad_settings_formatting_for_entity_mapping(self, mocked_sg):
+        """Test all the use cases where the entity mappings setting is not correctly formatted."""
+
+        # "sg_entity" key must be defined in the entity_mapping dictionary
+        self.assertRaises(RuntimeError, self._get_syncer, mocked_sg, "entities_generic_bad_sg_entity_formatting")
```

Comment:
Feel free to disagree, but in general, I like the test to contain all the relevant data. I don't mind fixtures, but I suppose I tend to use them if I have generic data that is going to be used in lots of tests and then modify that data if I need to in the actual test function. 
`entities_generic_bad_sg_entity_formatting` for example, is only used here, so I would have started with the fixture data, cloned it, modified it to  break it, and then assert it.

I think it makes it more readable (I don't need to jump to other parts), and the important stuff is contained within this test which I feel means it's less likely that making a change else where will break or invalidate this test.

Reply:
because of the way the bridge and the tests are built right now, I don't think it's feasible unfortunately...
(remember when I was saying that I preffered to write JS tests rather than Python tests? this was part of the "why" 😸 )